src/foam/nanos/column/NestedPropertiesExpression.js

```diff
@@ -81,61 +81,118 @@ foam.CLASS({
-        return foam.mlang.Expressions.create().DOT(prop, this.returnPropExpr(prop, this.returnDotExprForNestedProperty(prop.of, propName, ++i)));
+        if ( ! prop ) return null;
+
+        expr = this.buildPropertyExpr(prop, expr);
+
+        if ( i === propName.length - 1 )
+          return expr;
+
+        return this.returnDotExprForNestedProperty(prop.of, propName, ++i, expr);
       },
       javaCode: `
         ClassInfo ci = of;
-        PropertyInfo p = (PropertyInfo) ci.getAxiomByName(propName[i]);
-        if ( i == propName.length - 1 ) {
-          return p;
-        }
+        PropertyInfo prop = (PropertyInfo) ci.getAxiomByName(propName[i]);
+
+        if ( prop == null ) return null;
+
+        Boolean isPropAReference = isPropertyAReference(ci, prop);
+        expr = buildPropertyExpr(prop, expr, isPropAReference);
+
+        if ( i == propName.length - 1 )
+          return expr;
         
-        StringBuilder sb = new StringBuilder("find");
         try {
+          StringBuilder sb = new StringBuilder("find");
           Class cls;
-          if ( p instanceof foam.core.AbstractFObjectPropertyInfo ) {
-            cls = p.getValueClass();
-          } else {
+          if ( ! isPropAReference )
+            cls = prop.getValueClass();
+          else {
             sb.setLength(4);
-            Method m = ci.getObjClass().getMethod(StringUtil.capitalize(p.getName()), foam.core.X.class);
+            sb.append(StringUtil.capitalize(prop.getName()));
+            Method m = ci.getObjClass().getMethod(sb.toString(), foam.core.X.class);
             cls = m.getReturnType();
           }
           ci = (ClassInfo) cls.getMethod("getOwnClassInfo").invoke(null);
-        } catch (Exception e) {
-          Logger logger = (Logger) getX().get("logger");
-          logger.error(e);
+        } catch( Throwable t ) {
           return null;
         }
-
-        return returnPropExpr(p, returnDotExprForNestedProperty(ci, propName, ++i));
+        
+        return returnDotExprForNestedProperty(ci, propName, ++i, expr);
       `
     },
     {
-      name: 'returnPropExpr',
+      name: 'buildPropertyExpr',
       javaType: 'foam.mlang.Expr',
       args: [
         {
-          name: 'prop1',
-          class: 'Object',
-          javaType: 'foam.mlang.Expr'
+          name: 'prop',
+          javaType: 'foam.core.PropertyInfo',
         },
         {
-          name: 'prop2',
-          class: 'Object',
+          name: 'expr',
           javaType: 'foam.mlang.Expr'
+        },
+        {
+          name: 'isPropertyAReference',
+          type: 'Boolean'
         }
       ],
-      code: function (prop1, prop2) {
-        return foam.mlang.Expressions.create().DOT(prop1, prop2);
+      code: function(prop, expr) {
+        if ( foam.core.Reference.isInstance(prop) ) {
+          prop = foam.mlang.Expressions.create().REF(prop);
+        }
+      
+        return expr == null ? prop :
```

Comment:
expr is `undefined` when not given
```suggestion
        return ! expr ? prop :
```

Reply:
fixed